ksql-common/src/main/java/io/confluent/ksql/util/KsqlConfig.java

```diff
@@ -451,13 +454,18 @@ private KsqlConfig(final ConfigGeneration generation, final Map<?, ?> props) {
             .defaultCacheMaxBytesBufferingConfig);
     streamsConfigDefaults.put(StreamsConfig.NUM_STREAM_THREADS_CONFIG, KsqlConstants
         .defaultNumberOfStreamsThreads);
-    final Object fail = originals().get(FAIL_ON_DESERIALIZATION_ERROR_CONFIG);
-    if (fail == null || !Boolean.parseBoolean(fail.toString())) {
+    if (!getBooleanConfig(FAIL_ON_DESERIALIZATION_ERROR_CONFIG, false)) {
       streamsConfigDefaults.put(
           StreamsConfig.DEFAULT_DESERIALIZATION_EXCEPTION_HANDLER_CLASS_CONFIG,
           LogMetricAndContinueExceptionHandler.class
       );
     }
+    if (!getBooleanConfig(FAIL_ON_PRODUCTION_ERROR_CONFIG, false)) {
+      streamsConfigDefaults.put(
+          StreamsConfig.DEFAULT_PRODUCTION_EXCEPTION_HANDLER_CLASS_CONFIG,
+          LogAndContinueProductionExceptionHandler.class
```

Comment:
I was thinking about this a bit more. I think it actually makes sense to log in either case, but have the exception handler choose to return either FAIL or CONTINUE based on the value of this config.